wdl-lint/tests/lints/missing-runtime-block/source.errors

```diff
@@ -0,0 +1,8 @@
+warning[missing_runtime]: task `bad` is missing a runtime section
```

Comment:
Is `<severity>[<ID>]` clear enough that ID is an _identifier_? Let's assume a user who has never encountered a lint warning before. (I think that's a real possibility with some biologists being introduced to programming with WDL)

Is the `[<ID>]` clear enough that _this is what to look up in the documentation_? I think it becomes clear if there's an additional message at the bottom saying `look at 'sprocket explain <ID>' for more info`, but that message doesn't belong as built into the `Diagnostic`. This problem is similarly avoided if we provide a URL. But without those two, I'm worried we're not being clear enough.

Reply:
I think the majority of users will see these messages in the context of `sprocket` and it will be pretty clear with the additional provided context ("see `sprocket explain`" and "here's the rule's URL for more information").

That said, I'm happy to make this more clear in usage outside of sprocket; right now we're piggy backing on `codespan-reporting`'s notion of `code` here, but we can format it anyway we like.

Possible alternative?

```
warning: task `bad` is missing a runtime section [lint: missing_runtime]
```

Reply:
> I think the majority of users will see these messages in the context of sprocket and it will be pretty clear with the additional provided context ("see sprocket explain" and "here's the rule's URL for more information").

This is true, so maybe it's unnecessary to make _this_ clearer. But it can't hurt, right?

IMO the `error`/`warning`/`note` distinction isn't really serving us at the diagnostic/output stage. (See [this comment](https://github.com/stjude-rust-labs/wdl/pull/68#discussion_r1630434927) for my argument on that)

So I'd recommend a format like:
```
[missing_runtime]: task `bad` is missing a runtime section
```
I think boxing it (with `[]`), putting it at the front, and the fact it's snake_cased, all work together to bring attention to the ID. Although... What if we PascalCased our IDs? I think starting with a capital letter could be a clearer demarkation that this is an ID of some sort.
```
[MissingRuntime]: task `bad` is missing a runtime section
```
PascalCase is looking more "official" or something to me. IDK the right words to describe it, but it has a better ~~~vibe~~~.

That said, I also see your suggestion (`[lint: missing_runtime]` at the end) working. That's pretty explicit that "this is the lint title/ID". If we agree to drop `warning`/`note` then I vote we go with my format. If we decide to keep `warning`/`note`, we can go with your format. I vote for PascalCasing the IDs either way though

Reply:
One argument that I might put forth to keep the severity prefix for the output is that my intention is to have both the validation rules and the lint rules run in a single pass over the AST when you do `sprocket lint` (conversely, `sprocket check` would do basically the same thing except it wouldn't add the lint rules to the validator).

Thus one might get a mix of errors (coming from the validation rules) and warnings (coming from the lint rules) in the output, so I think it'd be good to have the severity label present.

There's also something to be said about consistency with most source-parsing tools out there that fairly universally print a severity label before a message.

Reply:
I had not considered either of those points. I think you're right, let's keep the `Severity` prefix (and move the rule name to the end,)
```
warning: task `bad` is missing a runtime section [lint: MissingRuntime]
```
Thoughts on PascalCase?

Reply:
I don't have a preference for either, to be honest. Whatever folks think would look good both from a diagnostic output standpoint but also from a configuration standpoint (in a TOML/JSON/YAML file or in a source comment like `# allow: <id>`).

Reply:
I'm partial to PascalCase in all those contexts. In my mind that leading capital imparts a certain weight or importance that is good for a rule to have. Authority 🧑‍⚖️

Reply:
I've pushed up a change to pascal case and to also put the rule in the suffix rather than after the diagnostic.

Let's see what people think.